fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -162,6 +176,162 @@ public ParseNode visitExplain(StarRocksParser.ExplainContext context) {
+        List<PartitionDesc> partitionDesc = visit(context.rangePartitionDesc(), PartitionDesc.class);
+        return new RangePartitionDesc(
+                identifierList.stream().map(Identifier::getValue).collect(toList()),
+                partitionDesc);
+    }
+
+    @Override
+    public ParseNode visitSingleRangePartition(StarRocksParser.SingleRangePartitionContext context) {
+        PartitionKeyDesc partitionKeyDesc = (PartitionKeyDesc) visit(context.partitionKeyDesc());
+        return new SingleRangePartitionDesc(false, ((Identifier) visit(context.identifier())).getValue(),
+                partitionKeyDesc, null);
+    }
+
+    @Override
+    public ParseNode visitMultiRangePartition(StarRocksParser.MultiRangePartitionContext context) {
+        if (context.interval() != null) {
+            IntervalLiteral intervalLiteral = (IntervalLiteral) visit(context.interval());
+
+            return new MultiRangePartitionDesc(
+                    ((StringLiteral) visit(context.string(0))).getStringValue(),
+                    ((StringLiteral) visit(context.string(1))).getStringValue(),
+                    Long.parseLong(intervalLiteral.getValue().toString()),
+                    intervalLiteral.getTimeUnitIdent());
+        } else {
+            return new MultiRangePartitionDesc(
+                    ((StringLiteral) visit(context.string(0))).getStringValue(),
+                    ((StringLiteral) visit(context.string(1))).getStringValue(),
+                    Long.parseLong(context.INTEGER_VALUE().getText()));
+        }
+    }
+
+    @Override
+    public ParseNode visitPartitionKeyDesc(StarRocksParser.PartitionKeyDescContext context) {
+        PartitionKeyDesc partitionKeyDesc;
+        if (context.LESS() != null) {
+            List<PartitionValue> partitionValueList =
+                    visit(context.partitionValueList().get(0).partitionValue(), PartitionValue.class);
+            partitionKeyDesc = new PartitionKeyDesc(partitionValueList);
+        } else {
+            List<PartitionValue> lowerPartitionValueList =
+                    visit(context.partitionValueList().get(0).partitionValue(), PartitionValue.class);
+            List<PartitionValue> upperPartitionValueList =
+                    visit(context.partitionValueList().get(1).partitionValue(), PartitionValue.class);
+            partitionKeyDesc = new PartitionKeyDesc(lowerPartitionValueList, upperPartitionValueList);
+        }
+        return partitionKeyDesc;
+    }
+
+    @Override
+    public ParseNode visitPartitionValue(StarRocksParser.PartitionValueContext context) {
+        if (context.MAXVALUE() != null) {
+            return PartitionValue.MAX_VALUE;
+        } else {
+            return new PartitionValue(((StringLiteral) visit(context.string())).getStringValue());
+        }
+    }
+
+    @Override
+    public ParseNode visitDistributionDesc(StarRocksParser.DistributionDescContext context) {
+        int buckets = 10;
```

Comment:
What does this magic number means?

Reply:
default buckets number same as old parser